code/sound/ds.cpp

```diff
@@ -1024,6 +1028,25 @@ void ds_init_buffers()
 }
 
 /**
+* Check if the player is using OpenAL Soft,
+* which is required to use enhanced sound.
+* Returns true on success, false otherwise.
+*/
+bool ds_check_for_openal_soft()
+{	
+	const ALchar * renderer = alGetString(AL_RENDERER);
+	Assertion(renderer!= NULL, "ds_check_for_openal_soft: renderer is null!");
```

Comment:
Assertions should be used for programming errors but if the OpenAL implementation returns something invalid here it's a recoverable data error so I think it's better to just print a message to the log.